pkg/metrics/instances.go

```diff
@@ -0,0 +1,105 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"go.opentelemetry.io/otel/attribute"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+var groups = make(map[string]*instancesCounter)
+
+type instancesCounter struct {
+	Labels []attribute.KeyValue
+	Count  int64
+}
+type instancesMetric struct {
+	client client.Client
+	groups map[string]*instancesCounter
+}
+
+func newInstancesMetric(client client.Client) *instancesMetric {
+	return &instancesMetric{
+		client: client,
+		groups: make(map[string]*instancesCounter),
+	}
+}
+
+func (i *instancesMetric) Setup(ctx context.Context) error {
+	tracer := otel.GetTracerProvider().Tracer(v1.CustomMetricsTracer)
+	ctx, span := tracer.Start(ctx, "setup-jaeger-instances")
+	defer span.End()
+	meter := global.Meter(meterName)
+	_, err := meter.NewInt64ValueObserver("operator_jaeger_instances", i.callback,
```

Comment:
With which frequency is the callback called?

Reply:
IIRC this is called per collection interval.